### PR TITLE
Icsaas 406 change api endpoint to new urls

### PR DESCRIPTION
**Pull request by igorsimovski**

### CODE REVIEW

File: core/src/main/java/com/valtech/aem/saas/core/http/client/DefaultSearchAdminRequestExecutorService.java [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+                .map(httpHost -> {
+                    BasicCredentialsProvider basicCredentialsProvider = new BasicCredentialsProvider();
+                    AuthScope authScope = new AuthScope(
+                            httpHost);
+                    UsernamePasswordCredentials usernamePasswordCredentials = new UsernamePasswordCredentials(
+                            configuration.searchAdminRequestExecutorService_basicAuthentication_user(),
+                            configuration.searchAdminRequestExecutorService_basicAuthentication_password());
+                    log.debug(
+                            "Creating basic credentials provider with authScope: {}, user: {}",
+                            authScope,
```
Comment (nhirrle) [inline core/src/main/java/com/valtech/aem/saas/core/http/client/DefaultSearchAdminRequestExecutorService.java:132]: formatting looks weired here, can you please fix ?

**Comment by nhirrle** [comment]
we got duplicated code now. could you add some abstraction or delegation on top to reuse classes. also sonar fails due too that

File: core/src/main/java/com/valtech/aem/saas/core/http/client/SearchServiceConnectionConfigurationService.java [hunk 2]
```diff
@@ -5,47 +5,47 @@
...
-     */
-    String getBasicAuthenticationUser();
-
-    /**
-     * Gets the password for basic authentication.
-     *
-     * @return password.
-     */
-    String getBasicAuthenticationPassword();
+//    /**
```
Comment (nhirrle) [inline core/src/main/java/com/valtech/aem/saas/core/http/client/SearchServiceConnectionConfigurationService.java:8]: is ths commented on purpose?

File: core/src/main/java/com/valtech/aem/saas/core/http/client/SearchServiceConnectionConfigurationService.java [hunk 3]
```diff
@@ -5,47 +5,47 @@
...
+//     */
+//    String getBaseUrl();
 
-    /**
-     * Checks whether Basic Authentication is enabled.
-     *
-     * @return {@code true} if enabled.
-     */
-    boolean isBasicAuthenticationEnabled();
+//    /**
```
Comment (nhirrle) [inline core/src/main/java/com/valtech/aem/saas/core/http/client/SearchServiceConnectionConfigurationService.java:15]: is ths commented on purpose?

**Review by nhirrle — CHANGES_REQUESTED** [review]

File: core/src/main/java/com/valtech/aem/saas/core/http/client/DefaultSearchAdminRequestExecutorService.java [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+           immediate = true)
+@ServiceDescription("Search as a Service - Search Admin Request Executor Service")
+@Designate(ocd = DefaultSearchAdminRequestExecutorService.Configuration.class)
+public class DefaultSearchAdminRequestExecutorService implements SearchAdminRequestExecutorService {
+
+    @Reference
+    private HttpClientBuilderFactory httpClientBuilderFactory;
+
+    @Getter(value = AccessLevel.PRIVATE,
+            onMethod_ = {@Synchronized})
```
Comment (nhirrle) [inline core/src/main/java/com/valtech/aem/saas/core/http/client/DefaultSearchAdminRequestExecutorService.java:33]: what is this annotation synchronized doing?

File: core/src/main/java/com/valtech/aem/saas/core/http/client/DefaultSearchAdminRequestExecutorService.java [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+    @ObjectClassDefinition(name = "Search as a Service - Search Admin Request Executor Service Configuration",
+                           description = "URL and authentication details for connect to Search Admin Endpoint.")
+    public @interface Configuration {
+
+        String DEFAULT_WEB_SERVICE_URL = "https://search-admin.infocentric.swiss/admin";
+        boolean DEFAULT_BASIC_AUTHENTICATION_ENABLE = false;
+        boolean DEFAULT_JWT_AUTHENTICATION_ENABLE = false;
+
+        @AttributeDefinition(name = "Base URL",
+                             description = "The protocol + url for the search service") String searchAdminRequestExecutorService_baseurl() default DEFAULT_WEB_SERVICE_URL; // NOSONAR
```
Comment (nhirrle) [inline core/src/main/java/com/valtech/aem/saas/core/http/client/DefaultSearchAdminRequestExecutorService.java:102]: could you add a linebreak after the annotation definitions? this for all occurances in here. It should improve readability
```
@AttributeDefinition(name = "Base URL",
                             description = "The protocol + url for the search service") 
String searchAdminRequestExecutorService_baseurl() default DEFAULT_WEB_SERVICE_URL; // NOSONAR
```

File: core/src/main/java/com/valtech/aem/saas/core/http/client/DefaultSearchAdminRequestExecutorService.java (hunk 4 above)
Comment (igorsimovski) [reply core/src/main/java/com/valtech/aem/saas/core/http/client/DefaultSearchAdminRequestExecutorService.java:33]: It is similar to adding the **synchronized** modifier. According to lombok docs (https://projectlombok.org/features/Synchronized), it is a safer variant.

**Review by nhirrle — APPROVED** [review]
Some minor changes I left with comment but I am happy with the PR, therfor approving so you can merge after applying them

**Pull request merged**